### PR TITLE
Require --force or --picky when verifying to a different version

### PR DESCRIPTION
**Pull request by matthewrsj**
When running verify --fix with -m specifying a different version than
the current OS version, require --picky to enforce management of files
under /usr or require --force as an override. Adding --picky can solve
issues such as issue #238 where fixing backward caused glibc library
confusion.

Add functional tests as well.

Signed-off-by: Matthew Johnson <matthew.johnson@intel.com>

Addresses #238

### CODE REVIEW

File: src/helpers.c [hunk 1]
```diff
@@ -995,3 +995,13 @@ bool is_compatible_format(int format_num)
...
 }
+
+/* Check if the version argument is the same version as the current OS */
+bool is_current_version(int version)
+{
+	if (version < 0) {
+		return false;
+	}
+
+	return (version == get_current_version(path_prefix));
```
Comment (phmccarty) [inline src/helpers.c]: Can `get_current_version()` return an error at this point in the code? If so, I would handle its error before deciding how to return from this function.

File: src/verify.c [hunk 2]
```diff
@@ -719,6 +719,22 @@ int verify_main(int argc, char **argv)
 		}
 	}
 
+	/* If --fix was specified but --force and --picky are not present, check
+	 * that the user is only fixing to their current version. If they are
+	 * fixing to a different version, print an error message that they need to
+	 * specify --force or --picky. */
+	if (cmdline_option_fix && !is_current_version(version)) {
```
Comment (phmccarty) [inline src/verify.c]: Do you want to add special handling for `verify --install` as well?

File: src/helpers.c (hunk 1 above)
Comment (matthewrsj) [reply src/helpers.c]: well at this point get_current_version can return a -1, but version cannot be -1. So if get_current_version encounters an error (version == get_current_version(...)) will evaluate to false.

File: src/verify.c (hunk 2 above)
Comment (matthewrsj) [reply src/verify.c]: yes.

Comment (matthewrsj) [reply src/verify.c]: Actually, per our meatspace convo, the --install option should be fixed up outside this PR, and should not require --force or --picky to be run, and possible made into its own sub-command.

File: src/helpers.c (hunk 1 above)
Comment (phmccarty) [reply src/helpers.c]: Okay, sounds good. I was lacking context of the surrounding code, so I think you can keep it as is.

File: src/verify.c (hunk 2 above)
Comment (phmccarty) [reply src/verify.c]: Agreed. Let's not special-case --install in this case for now.

**Comment by matthewrsj** [comment]
@phmccarty any other comments?

**Comment by phmccarty** [comment]
Nope, +1 from me.

**Pull request merged**